src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls

```diff
@@ -0,0 +1,309 @@
+     */
+    @future
+    public static void saveDeploymentId(String deploymentId) {
+        Data_Import_Settings__c dataImportSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
+        dataImportSettings.CMT_Deployment_ID__c = deploymentId;
+
+        try {
+            upsert dataImportSettings;
+        } catch (Exception e) {
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR,
+                Label.bdiErrorDataImportCustomSettings + ' ' + e.getMessage()));
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Action Method used by Poller to check status of CMDT deployment
+    * @return void
+    */
+    public PageReference startPolling() {
+        timeoutCount++;
+
+        refreshCustomSettings();
+        setDeploymentStatus();
+
+        if (deploymentResult != null) {
+            // We had a migration that contained CMT records to insert
+            Boolean isSuccess = deploymentResult.status == Metadata.DeployStatus.Succeeded
+                && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING;
+
+            if (isSuccess) {
+                isPolling = false;
+                successMessage();
+            } else if (deploymentResult.status == Metadata.DeployStatus.FAILED) {
+                isPolling = false;
+                addPageMessage(ApexPages.Severity.ERROR, Label.stgDeploymentTryAgain);
+            }
+
+        } else if (deploymentResult == null
+                   && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING) {
+            // We had a migration that didn't contain CMT records to insert
+            // DeploymentContainer was empty which means we're using our standard field mapping set
+            // Org had no conflicting field mappings nor any net new field mappings in their Data Import help text
+            isPolling = false;
+            successMessage();
+        }
+
+        if (timeoutCount > MAX_TIME_OUT && isPolling == true) {
+            isPolling = false;
+            addPageMessage(ApexPages.Severity.WARNING,
+                Label.stgDeploymentTimedOut + ' ' + Label.stgDeploymentTryAgain);
+            return null;
+        }
+
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description Display successful enablement of Advanced Mapping based on if there were some invalid
+    * help text mappings
+    */
+    private void successMessage() {
+        if (migrationMappingUtility.badHelpTextMappings.size() > 0) {
+            addPageMessage(ApexPages.Severity.WARNING,
+                'Custom Metadata mapping is enabled, but we found some invalid help text mappings that we couldn\'t migrate. Please look at the table below.');
```

Comment:
Turn to custom label @melissabarber

Reply:
Advanced Mapping is enabled, but we found some invalid Help Text mappings that couldn't migrate. *Take a screenshot of the table below, it will not be displayed when you leave this page.*